src/order/preorder_hom.lean

```diff
@@ -146,3 +146,23 @@ instance {β : Type*} [complete_lattice β] : complete_lattice (α →ₘ β) :=
   .. (_ : order_bot (α →ₘ β)) }
 
 end preorder_hom
+
+namespace rel_embedding
+
+variables {α β : Type*} [partial_order α] [partial_order β]
+variables (f : ((<) : α → α → Prop) ↪r ((<) : β → β → Prop))
+
+/-- A map between partial orders that is strictly monotonic is weakly monotonic. -/
+@[simps] def to_preorder_hom : α →ₘ β :=
+{ to_fun    := f,
+  monotone' := λ a b h, by
+  { by_cases hab : a = b,
+    { rw hab, },
+    { rw [ne.le_iff_lt hab, f.map_rel_iff] at h, exact le_of_lt h, }, }, }
+
+lemma to_preorder_hom_apply (a : α) : f.to_preorder_hom a = f a := rfl
```

Comment:
It might make sense to express this without the `a` argument, using `\u=`

Reply:
Thanks, now done!